### PR TITLE
Fix Go discovery bugs

### PR DESCRIPTION
**Pull request by saicheems**

### CODE REVIEW

File: src/test/java/com/google/api/codegen/testdata/discoveries/go/go_appengine.v1beta5.json.baseline [hunk 1]
```diff
@@ -737,7 +737,7 @@ func main() {
     // TODO: Add the desired fields to be changed.
   }
 
-  resp, err := s.Apps.Services.Versions.Patch(appsId, servicesId, versionsId, &version).Context(ctx).Do()
+  resp, err := s.Apps.Services.Versions.Patch(appsId, servicesId, versionsId, version).Context(ctx).Do()
```
Comment (broady) [inline src/test/java/com/google/api/codegen/testdata/discoveries/go/go_appengine.v1beta5.json.baseline]: Can we map these names to `appsID`? no big deal if not.

**Comment by saicheems** [comment]
Sorry I think I reuploaded while you were reviewing

Removed `fmt` in favor of `log` in every case, and renamed the request body to `body` to avoid some frequent name clashes.

**Comment by saicheems** [comment]
As for the `Id` thing... It's worth making note of

It's a bit tricky to detect acronyms, but we might be able to special case common ones.

**Comment by broady** [comment]
yeah, don't bother with the `Id` thing.

**Comment by saicheems** [comment]
Renamed `body` to `resource`

**Comment by broady** [comment]
LGTM

**Pull request merged**